src/options.ts

```diff
@@ -45,14 +45,21 @@ type SpanProcessorFactory = (
 
 type PropagatorFactory = (options: Options) => TextMapPropagator;
 
+export enum TracesExporter {
+  JAEGER = 'jaeger',
+  JAEGER_THRIFT_SPLUNK = 'jaeger-thrift-splunk',
+  OTLP = 'otlp',
+}
+
 export interface Options {
-  endpoint: string;
+  endpoint?: string;
   serviceName: string;
   accessToken: string;
   maxAttrLength: number;
   serverTimingEnabled: boolean;
   logInjectionEnabled: boolean;
   instrumentations: InstrumentationOption[];
+  tracesExporter: TracesExporter;
```

Comment:
we already have a spanExporterFactory option below, do we really need this?

Reply:
We could live without it, but I think it's useful to have as much parity between different formats of configuration as possible. Sure, we cannot let people set functions through env vars which is why I'm open to removing this if you'd really prefer that and have it conceptually be a alternative to setting the function in "`config.js`" BUT through environment. WDYT?

Reply:
I feel this is cluttering the constructor unnecessarily. Now we are providing two different ways of configuring an exporter with code. What if both are defined? Which one takes precedence? How do users decide which one to use when? If we accept tracesExporter as a well-known string, why not processor as well? What happens when we add metrics support? Do we add two options per component again?  This raises too many questions and doesn't solve enough problems (IMO) to be worth it. I think we should remove it and continue to use factory pattern.

Reply:
Fair. Removed it. In doing so, introducing `otlp-http` alias became trivial, so I did that as well.